### PR TITLE
Add bar chart

### PR DESCRIPTION
**Pull request by ZeeshaanMaudar**
### Scope
Add BarChart into view and update to use typescript

### CODE REVIEW

File: src/components/views/DepartmentPreview/index.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ class Preview extends Component<Tprops, Tstate> {
     super(props);
 
     const { initialSelected } = this.props;
-
+    console.log(mock);
```
Comment (schalkventer) [inline src/components/views/DepartmentPreview/index.tsx]: Remove the console log and import of `mock`.

File: src/helpers/utilities/trimValues/index.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+const trimValues = (rawValue, abbreviated) => {
```
Comment (schalkventer) [inline src/helpers/utilities/trimValues/index.js]: Convert to `.ts` file.

File: src/components/views/DepartmentPreview/BarChart/styled.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import styled from 'styled-components';
```
Comment (schalkventer) [inline src/components/views/DepartmentPreview/BarChart/styled.js]: Convert to `.ts` file

File: src/components/views/DepartmentPreview/BarChart/addRatios.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { maxBy } from 'lodash';
```
Comment (schalkventer) [inline src/components/views/DepartmentPreview/BarChart/addRatios.js]: Convert to `.ts` file

File: src/components/views/DepartmentPreview/BarChart/Bar/styled.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
+import styled from 'styled-components';
```
Comment (schalkventer) [inline src/components/views/DepartmentPreview/BarChart/Bar/styled.js]: Convert to `.ts` file

File: src/components/views/DepartmentPreview/BarChart/Bar/generateColor.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import { colors } from './data.json';
```
Comment (schalkventer) [inline src/components/views/DepartmentPreview/BarChart/Bar/generateColor.js]: convert to `.ts` file

File: src/components/views/DepartmentPreview/BarChart/Bar/ResizeWindowListener.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
+import { debounce } from 'lodash';
```
Comment (schalkventer) [inline src/components/views/DepartmentPreview/BarChart/Bar/ResizeWindowListener.js]: convert to `.ts` file

File: src/components/views/DepartmentPreview/index.tsx (hunk 1 above)
Comment (ZeeshaanMaudar) [reply src/components/views/DepartmentPreview/index.tsx]: done

File: src/helpers/utilities/trimValues/index.js (hunk 2 above)
Comment (ZeeshaanMaudar) [reply src/helpers/utilities/trimValues/index.js]: done

File: src/components/views/DepartmentPreview/BarChart/styled.js (hunk 3 above)
Comment (ZeeshaanMaudar) [reply src/components/views/DepartmentPreview/BarChart/styled.js]: done

File: src/components/views/DepartmentPreview/BarChart/addRatios.js (hunk 4 above)
Comment (ZeeshaanMaudar) [reply src/components/views/DepartmentPreview/BarChart/addRatios.js]: issues here

File: src/components/views/DepartmentPreview/BarChart/Bar/styled.js (hunk 5 above)
Comment (ZeeshaanMaudar) [reply src/components/views/DepartmentPreview/BarChart/Bar/styled.js]: done

File: src/components/views/DepartmentPreview/BarChart/Bar/generateColor.js (hunk 6 above)
Comment (ZeeshaanMaudar) [reply src/components/views/DepartmentPreview/BarChart/Bar/generateColor.js]: possible issue

File: src/components/views/DepartmentPreview/BarChart/Bar/ResizeWindowListener.js (hunk 7 above)
Comment (ZeeshaanMaudar) [reply src/components/views/DepartmentPreview/BarChart/Bar/ResizeWindowListener.js]: done

File: src/components/views/DepartmentPreview/BarChart/Bar/generateColor.js (hunk 6 above)
Comment (ZeeshaanMaudar) [reply src/components/views/DepartmentPreview/BarChart/Bar/generateColor.js]: done

File: src/components/views/DepartmentPreview/BarChart/addRatios.js (hunk 4 above)
Comment (ZeeshaanMaudar) [reply src/components/views/DepartmentPreview/BarChart/addRatios.js]: done

**Pull request merged**